### DISCUSSION TITLE
Better URL for shared models

### DISCUSSION THREAD

**Discussion by karimi** [clicketyclackety/Crash#31] [Ideas]
When starting a shared model, it'll be nice to have a URL that you can send to someone and that's all they need. Clicking that URL should open Rhino and either:

- Open packageManager if Crash is not installed

or 

- Open the shared model if Crash is installed

I imagine it looks something like this:
[rhino://package/search?q=Crash&host=0.0.0.0:5000](rhino://package/search?q=Crash&host=0.0.0.0:5000)

The URL above actually opens Rhino and searches package manager for Crash. The part that I haven't figured out is how to run `OpenSharedModel` upon launch and supply the `host` parameter from the share url.

**Comment by clicky** [comment]
That is absolutely incredible. I had no idea. Being able to jump into a model from a URL would be amazing.
How did you find this out or know this @karimi?

**Reply by karimi** [reply]
Well we do launch rhino from the web browser sometimes after logging in. I though it'll be nice to use the same mechanism here.
However right now Rhino only accepts certain types of URLs like the one above that opens the packageManager. We probably can make a change to Rhino to handle URLs to launch a plugin and pass the arguments. I'm going to take a stab at it

**Reply by clicky** [reply]
That sounds very powerful

**Reply by cwensley** [reply]
This is cool!  Do note that you'll have to look up the actual IP address of the machine and decide whether you want the external IP or the local IP.. then there's the problem of if they're connected to WiFi and Ethernet at the same time.   For external IP's behind a router there are some libraries we can use to poke open a port, but this would likely only work in a home setting vs. an office.

**Reply by clicky** [reply]
If it was a server hosted instance like the azure one, this would work out of the box
